domain/machine/state/state.go

```diff
@@ -1267,6 +1268,121 @@ WHERE machine_uuid = $machineUUID.uuid;
 	return decodeConstraints(result), nil
 }
 
+// GetModelConstraints returns the currently set constraints for the model.
+// The following error types can be expected:
+// - [modelerrors.NotFound]: when no model exists to set constraints for.
+// - [modelerrors.ConstraintsNotFound]: when no model constraints have been
+// set for the model.
+// Note: This method should mirror the model domain method of the same name.
+func (st *State) GetModelConstraints(
```

Comment:
Looks like we don't have any tests for this?

Reply:
Done!, nice catch.